types/src/check_summed_hex.rs

```diff
@@ -57,7 +61,8 @@ pub fn encode(input: &(impl AsRef<[u8]> + ?Sized)) -> String {
     let mut hex_output_string = String::with_capacity(input_bytes.len() * 2);
     for nibble in bytes_to_nibbles(input_bytes) {
         let c = HEX_CHARS[nibble as usize];
-        if c.is_alphabetic() && hash_bits.next().unwrap_or(true) {
+        let hash_bit = hash_bits.next().unwrap_or(true);
+        if c.is_alphabetic() && hash_bit {
```

Comment:
The old code deliberately did not advance the iterator if `c` is a number, because then we don't need the bit.
That way we always make use of as many bits from the hash as possible.